includes/class-wcsatt-product.php

```diff
@@ -0,0 +1,856 @@
+	 * - null    to indicate that the susbcription state of the product is undefined.
+	 *
+	 * @param  WC_Product  $product  Product object.
+	 * @param  string      $key      Identifier of subscription scheme to activate on object.
+	 * @return boolean               Action result.
+	 */
+	public static function set_subscription_scheme( $product, $key ) {
+
+		$active_scheme_key = self::get_subscription_scheme( $product );
+		$schemes           = self::get_subscription_schemes( $product );
+		$scheme_set        = false;
+
+		if ( ! empty( $key ) && is_array( $schemes ) && isset( $schemes[ $key ] ) && $key !== $active_scheme_key ) {
+
+			$scheme_to_set = $schemes[ $key ];
+
+			// Set subscription scheme key.
+			self::set_product_property( $product, 'active_subscription_scheme_key', $key );
+
+			/*
+			 * Set subscription scheme details.
+			 *
+			 * Note that prices are not set directly on object:
+			 * The price strings of many product types depend on more than the values returned by the abstract class price getters.
+			 * If we are going to apply filters anyway, there's no need to permanently set raw prices here.
+			 */
+			self::set_product_property( $product, 'subscription_period', $scheme_to_set[ 'subscription_period' ] );
+			self::set_product_property( $product, 'subscription_period_interval', $scheme_to_set[ 'subscription_period_interval' ] );
+			self::set_product_property( $product, 'subscription_length', $scheme_to_set[ 'subscription_length' ] );
+
+			$scheme_set = true;
+
+		} elseif ( empty( $key ) ) {
+
+			// Reset subscription scheme key.
+			self::set_product_property( $product, 'active_subscription_scheme_key', false === $key ? false : null );
+
+			// Reset subscription scheme details.
+			self::set_product_property( $product, 'subscription_period', null );
+			self::set_product_property( $product, 'subscription_period_interval', null );
+			self::set_product_property( $product, 'subscription_length', null );
+
+			$scheme_set = true;
+		}
+
+		do_action( 'wcsatt_set_product_subscription_scheme', $key, $active_scheme_key, $product );
+
+		return $scheme_set;
+	}
+
+	/**
+	 * Set the product as purchasable on a recurring basis only.
+	 *
+	 * @param  WC_Product  $product                 Product object to set.
+	 * @param  boolean     $is_forced_subscription  Value.
+	 */
+	public static function set_forced_subscription( $product, $is_forced_subscription ) {
+		self::set_product_property( $product, 'has_forced_subscription', $is_forced_subscription ? 'yes' : 'no' );
+	}
+
+	/*
+	|--------------------------------------------------------------------------
+	| Helpers
+	|--------------------------------------------------------------------------
+	*/
+
+	/**
+	 * Property getter (compatibility wrapper).
+	 *
+	 * @param  WC_Product  $product   Product object.
+	 * @param  string      $property  Property name.
+	 * @return mixed
+	 */
+	private static function get_product_property( $product, $property ) {
```

Comment:
Agreed -- see https://github.com/Prospress/woocommerce-subscribe-all-the-things/pull/136#discussion_r125583798

As it stands, `get_runtime_meta` would be the right name for it, if we end up using it after all. Looks like we might need/want to take the discussion about these in a separate issue?